Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java

```diff
@@ -894,17 +893,28 @@ public void run() {
         return used;
     }
 
+    public static Inventory getTopInventory(InventoryEvent event) {
+        try {
+            Object view = event.getView();
+            Method getTopInventory = view.getClass().getMethod("getTopInventory");
```

Comment:
Expensive reflection should be done once. Just call invoke in this method.

Reply:
Like this? Or would it be better to solely declare the variable outside the method and change the declaration inside the method to an assignment
```java
    private static Method getTopInventoryMethod;

    static {
        try {
            getTopInventoryMethod = InventoryView.class.getMethod("getTopInventory");
            getTopInventoryMethod.setAccessible(true);
        } catch (NoSuchMethodException e) {
            throw new RuntimeException(e);
        }
    }

    public static Inventory getTopInventory(InventoryEvent event) {
        try {
            Object view = event.getView();
            return (Inventory) getTopInventoryMethod.invoke(view);
        } catch (InvocationTargetException | IllegalAccessException e) {
            throw new RuntimeException(e);
        }
    }
```

Reply:
I'd shorten further by just calling `invoke(event.getView())`

Reply:
does the method need to have its accessibility explicitly set? it should be public and is fetched with `getMethod`, which fetches only public members - it should already be accessible